openssl-sys/src/handwritten/x509_sbgp.rs

```diff
@@ -0,0 +1,222 @@
+#[cfg(ossl110)]
+pub union ASIdentifierChoice_st_anon_union {
+    pub asIdsOrRanges: *mut stack_st_ASIdOrRange,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct ASIdentifierChoice {
+    pub type_: c_int,
+    pub u: ASIdentifierChoice_st_anon_union,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct ASIdentifiers {
+    pub asnum: *mut ASIdentifierChoice,
+    pub rdi: *mut ASIdentifierChoice,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct IPAddressRange {
+    pub min: *mut ASN1_BIT_STRING,
+    pub max: *mut ASN1_BIT_STRING,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct IPAddressOrRange {
+    pub type_: c_int,
+    pub u: IPAddressOrRange_st_anon_union,
+}
+#[repr(C)]
+#[cfg(ossl110)]
+pub union IPAddressOrRange_st_anon_union {
+    pub addressPrefix: *mut ASN1_BIT_STRING,
+    pub addressRange: *mut IPAddressRange,
+}
+
+#[cfg(ossl110)]
+stack!(stack_st_IPAddressOrRange);
+#[cfg(ossl110)]
+type IPAddressOrRanges = stack_st_IPAddressOrRange;
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub union IPAddressChoice_st_anon_union {
+    pub addressesOrRanges: *mut IPAddressOrRanges,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct IPAddressChoice {
+    pub type_: c_int,
+    pub u: IPAddressChoice_st_anon_union,
+}
+
+#[repr(C)]
+#[cfg(ossl110)]
+pub struct IPAddressFamily {
+    pub addressFamily: *mut ASN1_OCTET_STRING,
+    pub ipAddressChoice: *mut IPAddressChoice,
+}
+
+#[cfg(ossl110)]
+stack!(stack_st_IPAddressFamily);
+#[cfg(ossl110)]
+type IPAddrBlocks = stack_st_IPAddressFamily;
+
+#[cfg(ossl110)]
+extern "C" {
+    pub fn ASIdentifiers_free(asi: *mut ASIdentifiers);
+    pub fn ASIdOrRange_free(asi: *mut ASIdOrRange);
+    pub fn IPAddressFamily_free(asi: *mut IPAddressFamily);
+    pub fn IPAddressOrRange_free(asi: *mut IPAddressOrRange);
+}
+
+#[cfg(ossl110)]
+pub unsafe fn X509v3_addr_get_afi(f: *mut IPAddressFamily) -> c_int {
+    if f.is_null() {
+        0
+    } else {
+        let d = (*f).addressFamily as *mut ASN1_STRING;
+        if d.is_null() || ASN1_STRING_length(d) < 2 || ASN1_STRING_get0_data(d).is_null() {
+            0
+        } else {
+            let raw = ASN1_STRING_get0_data(d);
+            ((*raw.offset(0) as c_int) << 8) | *raw.offset(1) as c_int
+        }
+    }
+}
+
+#[cfg(ossl110)]
+fn length_from_afi(afi: c_int) -> isize {
+    match afi {
+        IANA_AFI_IPV4 => 4,
+        IANA_AFI_IPV6 => 16,
+        _ => 0,
+    }
+}
+
+#[cfg(ossl110)]
+struct ASN1_STRING_internal {
+    length: c_int,
+    type_: c_int,
+    data: *mut u8,
+    /*
+     * The value of the following field depends on the type being held.  It
+     * is mostly being used for BIT_STRING so if the input data has a
+     * non-zero 'unused bits' value, it will be handled correctly
+     */
+    flags: c_int,
+}
+
+/*
+ * Expand the bitstring form of an address into a raw byte array.
+ * At the moment this is coded for simplicity, not speed.
+ */
+#[cfg(ossl110)]
+fn addr_expand(addr: *mut u8, bs: *const ASN1_BIT_STRING, length: isize, fill: u8) -> bool {
```

Comment:
This doesn't seem like something that should be in openssl-sys.

Reply:
This function, and associated functions , has now been replaced with a FFI binding to `X509v3_addr_get_range` and `X509v3_addr_get_afi`.